js/settingsHelpers.js

```diff
@@ -367,43 +419,169 @@ export const changeMailingAddress = async (addressLine1, addressLine2, city, sta
+ * @param {String} email - the new email address 
+ * @returns {boolean} - true if the update was successful, false otherwise
+ */
+const updateFirebaseAuthEmail = async (firebaseAuthUser, email) => {
+  try {
+    await firebaseAuthUser.updateEmail(email);
+    return true;
+  } catch (error) {
+    handleUpdatePhoneEmailErrorInUI('updateFirebaseAuthEmail()', error);
+    return false;
+  }
+};
+
+/**
+ * Update the user's phone number in Firebase Auth.
+ * This walks the user through two verification processes:
+ *  (1) recaptcha verification in the browser.
+ *  (2) phone number verification with a texted code.
+ * @param {Object<User>} firebaseAuthUser - the firebase auth user object.
+ * @param {String} phone - the new cleaned phone number.
+ * @returns {boolean} - true if the update was successful, false otherwise.
+ * Note: '+1' has already been prepended to the phone number (format required by Firebase Auth).
+ */
+const updateFirebaseAuthPhone = async (firebaseAuthUser, phone) => {
+  try {
+      document.getElementById('changePhoneSubmit').style.display = 'none';
+      window.recaptchaVerifier = new firebase.auth.RecaptchaVerifier('recaptcha-container');
+
+      const recaptchaVerifier = window.recaptchaVerifier;
+      const provider = new firebase.auth.PhoneAuthProvider;
+
+      const verificationId = await provider.verifyPhoneNumber(phone, recaptchaVerifier);
+      const verificationCode = window.prompt('Please enter the verification code that was sent to your mobile device.');
+      if (!verificationCode) {
+          throw new Error("Verification code not provided");
+      }
+
+      const phoneCredential = firebase.auth.PhoneAuthProvider.credential(verificationId, verificationCode);
+      await firebaseAuthUser.linkWithCredential(phoneCredential);
+      window.recaptchaVerifier.clear();
+      return true;
+  } catch (error) {
+      console.error('updateFirebaseAuthPhone() error', error);
+      handleUpdatePhoneEmailErrorInUI('updateFirebaseAuthPhone()', error);
+      document.getElementById('recaptcha-container').style.display = 'none';
+      return false;
+  }
+};
+
+/**
+ * Unlink a provider from the current user's Firebase account.
+ * On Success, update the user's signInMechanism and (firebaseAuthEmail or firebaseAuthPhone) in the user's Firestore profile.
+ * @param {String} providerType - 'email' or 'phone' 
+ * @returns {boolean} - true if the unlink was successful, false otherwise
+ */
+export const unlinkFirebaseAuthProvider = async (providerType) => {
+  try {
+    const firebaseAuthUser = firebase.auth().currentUser;
+
+    if (!firebaseAuthUser) {
+      throw new Error('No user is currently authenticated');
+    }
+
+    if (!(providerType === 'email' || providerType === 'phone')) {
+      throw new Error('Invalid providerType. Expected "email" or "phone"');
+    }
+
+    const result = await unlinkFirebaseAuthenticationTrigger(providerType);
```

Comment:
Will `isSuccess` a better name for this boolean value?

Reply:
`isSuccess` is definitely better. Updated.